bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/FirstKeyOnlyFilterAdapter.java

```diff
@@ -29,7 +29,10 @@
  */
 public class FirstKeyOnlyFilterAdapter extends TypedFilterAdapterBase<FirstKeyOnlyFilter> {
 
-  private static RowFilter LIMIT_ONE = FILTERS.limit().cellsPerRow(1).toProto();
+  private static RowFilter LIMIT_ONE = FILTERS.chain()
+      .filter(FILTERS.value().strip())
+      .filter(FILTERS.limit().cellsPerRow(1))
```

Comment:
I'm not sure if it makes that much of a difference, but logically it would be more efficient to limit cells first